### PR TITLE
feat: Galaxy NFT

### PR DESCRIPTION
**Pull request by Chef-Cheems**

### CODE REVIEW

File: src/components/GlobalCheckClaimStatus/index.tsx [hunk 1]
```diff
@@ -29,17 +31,35 @@ const GlobalCheckClaimStatus: React.FC<GlobalCheckClaimStatusProps> = (props) =>
...
+  const [cid, setCid] = useState(null)
+  const [canClaimNFT, setCanClaimNFT] = useState(false)
   const { account } = useWeb3React()
   const { pathname } = useRouter()
-  const [onPresentAnniversaryModal] = useModal(<AnniversaryAchievementModal />)
+  const [onPresentModal] = useModal(<GalaxyNFTClaimModal cid={cid} />, false, true, 'galaxyNFTClaimModal')
 
   // Check claim status
   useEffect(() => {
     const fetchClaimAnniversaryStatus = async () => {
```
Comment (memoyil) [inline src/components/GlobalCheckClaimStatus/index.tsx:42]: We can change that function name as well

File: src/components/GlobalCheckClaimStatus/index.tsx [hunk 2]
```diff
@@ -29,17 +31,35 @@ const GlobalCheckClaimStatus: React.FC<GlobalCheckClaimStatusProps> = (props) =>
...
-      const { canClaim } = getAnniversaryAchievementContract()
-      const canClaimAnniversary = await canClaim(account)
-      setCanClaimAnniversaryPoints(canClaimAnniversary)
+      try {
+        const { campaign } = await request(
+          GALAXY_NFT_CLAIMING_API,
+          gql`
+            query checkEligabilityForGalaxyNFT($campaignId: ID!, $address: String!) {
```
Comment (memoyil) [inline src/components/GlobalCheckClaimStatus/index.tsx:47]: That typo is on their end right?

Comment (Chef-Cheems) [reply src/components/GlobalCheckClaimStatus/index.tsx]: nah, those are local query names, my typo - fixed, thanks

File: src/components/GlobalCheckClaimStatus/index.tsx [hunk 3]
```diff
@@ -29,33 +32,55 @@ const GlobalCheckClaimStatus: React.FC<GlobalCheckClaimStatusProps> = (props) =>
...
 
   // Check claim status
   useEffect(() => {
-    const fetchClaimAnniversaryStatus = async () => {
-      const { canClaim } = getAnniversaryAchievementContract()
-      const canClaimAnniversary = await canClaim(account)
-      setCanClaimAnniversaryPoints(canClaimAnniversary)
+    const fetchClaimNftStatus = async () => {
+      try {
+        const { campaign } = await request(
```
Comment (memoyil) [inline src/components/GlobalCheckClaimStatus/index.tsx:47]: We can put these two async calls in Promise.all

File: src/components/GlobalCheckClaimStatus/index.tsx [hunk 4]
```diff
@@ -29,33 +32,55 @@ const GlobalCheckClaimStatus: React.FC<GlobalCheckClaimStatusProps> = (props) =>
  */
 const GlobalCheckClaim: React.FC<GlobalCheckClaimStatusProps> = ({ excludeLocations }) => {
   const hasDisplayedModal = useRef(false)
-  const [canClaimAnniversaryPoints, setCanClaimAnniversaryPoints] = useState(false)
+  const [cid, setCid] = useState(null)
+  const [canClaimNFT, setCanClaimNFT] = useState(false)
+  const [nftBalance, setNftBalance] = useState(999) // default high to avoid fashing modal
```
Comment (memoyil) [inline src/components/GlobalCheckClaimStatus/index.tsx:37]: typo on comment

File: src/components/GlobalCheckClaimStatus/index.tsx (hunk 2 above)
Comment (memoyil) [reply src/components/GlobalCheckClaimStatus/index.tsx]: ah yes you are right 👍

File: src/components/GlobalCheckClaimStatus/index.tsx [hunk 5]
```diff
@@ -29,33 +32,55 @@ const GlobalCheckClaimStatus: React.FC<GlobalCheckClaimStatusProps> = (props) =>
...
   const hasDisplayedModal = useRef(false)
-  const [canClaimAnniversaryPoints, setCanClaimAnniversaryPoints] = useState(false)
+  const [cid, setCid] = useState(null)
+  const [canClaimNFT, setCanClaimNFT] = useState(false)
+  const [nftBalance, setNftBalance] = useState(999) // default high to avoid flashing modal
+  const galaxyNFTContract = useERC721('0x4344cDD59780CC7B8A47Cd903424F91f41A3a366')
   const { account } = useWeb3React()
   const { pathname } = useRouter()
-  const [onPresentAnniversaryModal] = useModal(<AnniversaryAchievementModal />)
+  const [onPresentModal] = useModal(<GalaxyNFTClaimModal cid={cid} />, false, true, 'galaxyNFTClaimModal')
```
Comment (memoyil) [inline src/components/GlobalCheckClaimStatus/index.tsx:41]: Close on overlay click cannot be disabled right now, there is a fix for that https://github.com/pancakeswap/pancake-frontend/pull/3814

Comment (Chef-Cheems) [reply src/components/GlobalCheckClaimStatus/index.tsx:41]: yeah its ok, I just need to put something in there to set up updating modal :)

File: src/components/GlobalCheckClaimStatus/index.tsx (hunk 3 above)
Comment (Chef-Cheems) [reply src/components/GlobalCheckClaimStatus/index.tsx:47]: nah, let's just release it, its not gonna be live for long time anyway 😂

**Review by chef-jojo — APPROVED** [review]

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**